frontend/components/TitleBar/TitleBar.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { bindActionCreators } from 'redux';
+import { View, Text } from 'react-native';
+import { styles } from './TitleBar.Styles';
+
+import Button from '../ButtonField/Button';
+
+import {
+    createConnectionData,
+    getAllConnectionDataAction,
+    getConnectionDataByRadiusAction,
+    showMap,
+} from '../ButtonField/ButtonField.Action';
+import { connect } from 'react-redux';
+
+class TitleBar extends React.Component {
+    renderTitle = () => {
+        const { showingMap } = this.props;
+        if (showingMap) {
+            return (
+                <View style={[styles.toolbarTitle, { alignItems: 'center' }]}>
+                    <Text style={styles.toolbarTitle}>
+                        Click on your location for further information.
+                    </Text>
+                </View>
+            );
+        }
+        return (
+            <Text style={styles.toolbarTitle}>
+                Your Current Device Information
+            </Text>
+        );
+    };
+
+    render() {
+        const { showingMap } = this.props;
+        return (
+            <View style={styles.toolbar}>
+                <Button
+                    onPress={() => this.props.showMap(!showingMap)}
+                    disabled={!showingMap}
+                    style={styles.toolbarButton}
+                    textStyle={styles.toolbarButtonText}
+                    title={showingMap ? 'Back' : ''}
+                />
+                {this.renderTitle()}
+                <Button
```

Comment:
If Button has no use, remove it instead of disabling it with an empty text.

Reply:
no changes here because of styling issues...